Do not init facade in adapter

henhuy opened the pull request:
Closes #46

henhuy committed:
Add empty collections folder to tests

henhuy committed:
Refactor Adapter to not init facade

henhuy commented:
For now I only added a simple test to check if my new solution works.
Other tests and implementations are not adopted and will fail

henhuy commented:
Now `Adapter` class looks very empty and could almost be removed (mapper inside Adapter does most parts).
Maybe next time we should refactor even more...

Commit:
Update lock file [data-adapter]

FelixMau commented on data_adapter_oemof/adapters.py:
```diff
@@ -14,34 +11,18 @@
-    @classmethod
-    def get_default_parameters(cls, struct: dict, mapper: Mapper) -> dict:
+    def as_dict(self) -> dict:
+        return self.facade_dict
```
Why do you want to keep the `as_dict` attribute? 
To me this appears a little confusing and like an extra step to just change the name.

henhuy replied:
only synthetic sugar. but i will remove it

FelixMau commented on data_adapter_oemof/adapters.py:
```diff
@@ -59,146 +40,118 @@ def get_default_parameters(cls, struct: dict, mapper: Mapper) -> dict:
     """
 
     type = "dispatchable"
+    facade = facades.Dispatchable
```
I think we have had this Version with facade as DataAdapter attribute before but I do not remember why we changed that.

FelixMau committed:
Update PROCESS_TYPE_MAP.yaml

No data_adapter test since it should be tested in data_adapter itself

FelixMau commented:
Merged the "fix tests" updates from dev into here for easier maintenance and development:

Todos:
- [ ] New Adapter class has no attribute "name" anymore (see mapping tests)
- [ ] New Adapter class has no "parametrize_dataclass" anymore ->replace by init the Adapter
- [ ] Resolve "as_dict" comment from above

FelixMau commented:
@henhuy Should I have a look into fixing the tests here or do you want to continue working on this issue?

henhuy commented:
Hey @FelixMau, would be nice if you could continue - as I'm deep into solph code, trying to implement TSAM... thx in advance!

FelixMau committed:
Dataclass dependencies

To create a Mapper instance a adapter is neccessary but to create an adapter instance a mapper is neccessary as well. Therefore a new function has been created in Mapper to do the job the adapter instance did before (giving out all names and types of occouring fields)

FelixMau commented:
To create a Mapper instance a `adapter` is neccessary but to create an `adapter` instance a mapper is [neccessary](https://github.com/sedos-project/data_adapter_oemof/blob/aee5bb011bd4cd1d1333e94ce47377cd0cec5b39/data_adapter_oemof/adapters.py#L23-L24) as well since we wanted to change `__init__` behaviour. 
This circular behavior might be problematic and is not good style I believe, we should discuss solutions if there is capacity. 

To solve this issue a new function has been created in Mapper to do the job the `adapter` instance did before (giving out all names and types of occouring fields)

Tests are still failing due to missing changes from dev. Should I merge dev into here?

henhuy commented:
I will have a look today!

henhuy committed:
Fix test mapping with sequence

henhuy committed:
Minor naming change

henhuy committed:
Minor change

henhuy commented:
> To create a Mapper instance a `adapter` is neccessary but to create an `adapter` instance a mapper is [neccessary](https://github.com/sedos-project/data_adapter_oemof/blob/aee5bb011bd4cd1d1333e94ce47377cd0cec5b39/data_adapter_oemof/adapters.py#L23-L24) as well since we wanted to change `__init__` behaviour. This circular behavior might be problematic and is not good style I believe, we should discuss solutions if there is capacity.

For `Mapper` instance only the class of `Adapter` has to be given (not an instance) - so IMO this is not a problem. (If instance would have been needed, you are right this would cause cyclic problems!)

> To solve this issue a new function has been created in Mapper to do the job the `adapter` instance did before (giving out all names and types of occouring fields)

Looks nice - good idea!  - I only renamed some names regarding your new namedtuple.

> Tests are still failing due to missing changes from dev. Should I merge dev into here?

I rebased dev into this branch.

henhuy commented:
By the way  - nice job of joining year-related processes!
Hope we can merge this soon. We are coming closer to our goal of building datapackage from databus data, I think.

We have to adapt "goal" CSVs, as non-given parameters are not set with default value from facade anymore by the adapter class (as they aren't necessary for setting up facade).

henhuy committed:
Add region and year to extra fields and remove additional attributes
